pkg/api/handlers/utils/handler.go

```diff
@@ -147,3 +151,103 @@ func GetDecoder(r *http.Request) *schema.Decoder {
 	}
 	return r.Context().Value(api.CompatDecoderKey).(*schema.Decoder)
 }
+
+// ParseJSONField unmarshals a JSON string into the target type with proper error formatting.
+func ParseJSONField[T any](jsonStr, fieldName string, target *T) error {
+	if err := json.Unmarshal([]byte(jsonStr), target); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ParseOptionalJSONField unmarshals a JSON string only if the field exists in query values.
+func ParseOptionalJSONField[T any](jsonStr, fieldName string, queryValues url.Values, target *T) error {
+	if _, found := queryValues[fieldName]; found {
+		return ParseJSONField(jsonStr, fieldName, target)
+	}
+	return nil
+}
+
+// ParseOptionalBool creates a types.OptionalBool if the field exists in query values.
+// Returns the OptionalBool and whether the field was found.
+func ParseOptionalBool(value bool, fieldName string, queryValues url.Values) (types.OptionalBool, bool) {
+	if _, found := queryValues[fieldName]; found {
+		return types.NewOptionalBool(value), true
+	}
+	var empty types.OptionalBool
```

Comment:
is it me or does this logic seem unusual?  i was thinking maybe `empty` should be declared first but then wondered if you did this on purpose for a hot path ?

Reply:
I declared empty later for the hot path and also realized this could be simplified by returning `types.OptionalBoolUndefined`.